src/nat/data_models/optimizable.py

```
@@ -65,6 +65,57 @@ def suggest(self, trial: Trial, name: str):
+
+        if self.step is None:
+            raise ValueError(
+                f"Grid search with range (low={self.low}, high={self.high}) requires 'step' to be specified. "
+                "Please define the step size to discretize the range, for example: step=0.1")
+
+        # Generate grid values from range with step
+        if isinstance(self.low, int) and isinstance(self.high, int):
+            step = int(self.step)
+            if self.log:
+                raise ValueError("Log scale is not supported for integer ranges in grid search. "
+                                 "Please use linear scale or provide explicit values.")
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)